pup-0002.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Docstring Style Conventions
+author: Austin Macdonald
+created: 28-Mar-2017
+status: proposed
+---
+
+## Summary
+
+This PUP proposes the use of Google style docstrings for Pulp 3+.
+
+## Motivation
+
+Most current Pulp developers find Google Style docstrings to be more readable than the default
+Sphinx style docstrings.
+
+## Details
+
+Upon acceptance, these changes will be documented in the Pulp style guide.
+
+### Style Choices
+
+Docstrings should follow the conventions documented in the
+[Google style guide](http://google.github.io/styleguide/pyguide.html?showone=Comments#Comments)
+except where Pulp specific additions or exceptions apply.
+
+Additionally, docstrings must be parsable by the
+[Napoleon plugin for Sphinx.](http://www.sphinx-doc.org/en/stable/ext/napoleon.html)
+
+### Additions/Exceptions
+
+Pulp conventions that differ from the `Comments` section of the Google style guide will be
+explicitly included in the Pulp style guide.
+
+1. Modules will not contain license information. Our license information is presented correctly
+within the project; it is unnecessary to include it in each file.
+
+2. The type of each `Arg` value should be included after the variable name in parentheses. The type of each `Returns` value should be the first item on the line.
+
+3. Following the type of `Arg` and `Return` values, there will be a colon and a single space
```

Comment:
The example on the napoleon docs shows exactly this. Is this an exception?
http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html